File: Source/Relativity.DataExchange.Legacy/AuthTokenProviderAdapter.vb
Imports kCura.WinEDDS
Imports Relativity.Transfer

''' This class represents method of refreshing the token credentials on the expiration event
''' Instance of this class is injected to Tapi
Public Class AuthTokenProviderAdapter
	Implements IAuthenticationTokenProvider

	Private ReadOnly _relativityTokenProvider As IRelativityTokenProvider

	Public Sub New(relativityTokenProvider As IRelativityTokenProvider)
		_relativityTokenProvider = relativityTokenProvider
	End Sub

	Public Function GenerateToken() As String Implements IAuthenticationTokenProvider.GenerateToken
		Return _relativityTokenProvider.GetToken()
	End Function

End Class

File: Source/Relativity.DataExchange.Core/AuthTokenProviderAdapter.cs
// <copyright file="AuthTokenProviderAdapter.cs" company="Relativity ODA LLC">
// © Relativity All Rights Reserved.
// </copyright>

namespace Relativity.DataExchange
{
	using Relativity.Transfer;

	/// <summary>
	/// This class represents method of refreshing the token credentials on the expiration event
	/// Instance of this class is injected to Tapi.
	/// </summary>
	internal class AuthTokenProviderAdapter : IAuthenticationTokenProvider
	{
		private readonly IRelativityTokenProvider _relativityTokenProvider;

		/// <summary>
		/// Initializes a new instance of the <see cref="AuthTokenProviderAdapter"/> class.
		/// </summary>
		/// <param name="relativityTokenProvider">IRelativityTokenProvider token instance.</param>
		public AuthTokenProviderAdapter(IRelativityTokenProvider relativityTokenProvider)
		{
			relativityTokenProvider.ThrowIfNull(nameof(relativityTokenProvider));

			this._relativityTokenProvider = relativityTokenProvider;
		}

		/// <summary>
		/// It delegated call to <see cref="IRelativityTokenProvider"/> method.
		/// </summary>
		/// <returns>Relativity authentication token value.</returns>
		public string GenerateToken()
		{
			return this._relativityTokenProvider.GetToken();
		}
	}
}
